include/Dialect/LWE/IR/LWETypes.td

```diff
@@ -0,0 +1,58 @@
+#ifndef INCLUDE_DIALECT_LWE_IR_LWETYPES_TD_
+#define INCLUDE_DIALECT_LWE_IR_LWETYPES_TD_
+
+include "include/Dialect/LWE/IR/LWEDialect.td"
+include "include/Dialect/LWE/IR/LWEAttributes.td"
+
+include "include/Dialect/Polynomial/IR/PolynomialAttributes.td"
+
+include "mlir/IR/DialectBase.td"
+include "mlir/IR/AttrTypeBase.td"
+
+// A base class for all types in this dialect
+class LWE_Type<string name, string typeMnemonic>
+    : TypeDef<LWE_Dialect, name> {
+  let mnemonic = typeMnemonic;
+}
+
+// LWE Ciphertexts are ranked tensors of integers representing the LWE samples
+// and the bias.
+def LWECiphertext : LWE_Type<"LWECiphertext", "lwe_ciphertext"> {
+  let summary = "A type for an LWE ciphertexts";
+
+  let description = [{
+    A type for LWE ciphertexts.
+
+    This type keeps track of the plaintext integer encoding for the LWE
+    Ciphertext to ensure proper decoding after decryption. It also keeps track
+    of the ring where the LWE ciphertext is defined, which provides information
+    on the ciphertext shape and the ring operations used in LWE operations.
+  }];
+
+  let parameters = (ins
+    LWE_BitFieldEncoding:$encoding,
+    Ring_Attr:$ring
```

Comment:
Here and in CGGI, why exactly do we need the ring? You want this type to support both LWE and RLWE? I feel they should be different types, but if we keep them as one type, we should really have the ring attribute be optional or constrained somehow.

Reply:
Oops! Ah! Yes, RLWE/LWE should be separate - this should instead have the cmod and dimension N of the lwe ciphertexts